### PR TITLE
Hw spring web rest was done

### PR DESCRIPTION
**Pull request by maks-untilov**

### CODE REVIEW

File: src/main/java/mate/academy/spring/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                           ShoppingCartService shoppingCartService, UserService userService,
+                           OrderService orderService) {
+        this.orderRequestDto = orderRequestDto;
+        this.orderResponseDto = orderResponseDto;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.orderService = orderService;
+    }
+
+    @PostMapping("/orders/complete")
```
Comment (Rommelua) [inline src/main/java/mate/academy/spring/controller/OrderController.java:39]: /orders - class level mapping

File: src/main/java/mate/academy/spring/controller/ShoppingCartController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            ShoppingCartService shoppingCartService,
+            MovieSessionService movieSessionService, UserService userService) {
+        this.shoppingCartResponseMapper = shoppingCartResponseMapper;
+        this.shoppingCartRequestMapper = shoppingCartRequestMapper;
+        this.shoppingCartService = shoppingCartService;
+        this.movieSessionService = movieSessionService;
+        this.userService = userService;
+    }
+
+    @PutMapping("/shopping-carts/movie-sessions")
```
Comment (Rommelua) [inline src/main/java/mate/academy/spring/controller/ShoppingCartController.java:38]: /shopping-carts - class level mapping

File: src/main/java/mate/academy/spring/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    private UserService userService;
+
+    @Autowired
+    public UserController(DtoResponseMapper<UserResponseDto, User> userResponseMapper,
+                          UserService userService) {
+        this.userResponseMapper = userResponseMapper;
+        this.userService = userService;
+    }
+
+    @GetMapping("/users/by-email")
```
Comment (Rommelua) [inline src/main/java/mate/academy/spring/controller/UserController.java:24]: same

File: src/main/java/mate/academy/spring/dao/impl/TicketDaoImpl.java [hunk 4]
```diff
@@ -11,4 +14,13 @@ public class TicketDaoImpl extends AbstractDao<Ticket> implements TicketDao {
     public TicketDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
+
+    @Override
+    public Optional<Ticket> get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.get(Ticket.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get Ticket by id, e", e);
```
Comment (Rommelua) [inline src/main/java/mate/academy/spring/dao/impl/TicketDaoImpl.java:23]: fix msg

File: src/main/java/mate/academy/spring/mapper/impl/request/OrderRequestMapper.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import mate.academy.spring.mapper.DtoRequestMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.Ticket;
+import mate.academy.spring.model.dto.request.OrderRequestDto;
+import mate.academy.spring.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class OrderRequestMapper implements DtoRequestMapper<OrderRequestDto, Order> {
```
Comment (Rommelua) [inline src/main/java/mate/academy/spring/mapper/impl/request/OrderRequestMapper.java:16]: image

**Review by Rommelua — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

image

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**